src/datatable.js

```diff
@@ -329,6 +330,19 @@ const dataTable = {
         // disable pagination to not press multiple on it
         disablePagination($elt.paginations);
 
+        if (typeof options.requestInterceptor === 'function') {
+            options
+                .requestInterceptor(parameters)
+                .then(data => {
+                    self._render($elt, data);
+                })
+                .catch(error => {
+                    $elt.trigger('error.' + ns, [error]);
+                    self._render($elt, {});
+                });
+            return;
```

Comment:
In this case not be invoked event and loading. Should be they invoked?
```
         /**
         * @event dataTable#query.datatable
         * @param {Object} ajaxConfig - The config object used to setup the AJAX request
         */
        $elt.trigger('query.' + ns, [ajaxConfig]);

        // display the loading state
        if (options.status) {
            $elt.find('.loading').removeClass(hiddenCls);
        }
```

Reply:
yeah, you're right it could be useful in case of `requestInterceptor` callback is handled a long task.
I'm going to move it right before AJAX call